### ISSUE TITLE
Inspector: "Object was deleted while awaiting a callback" on complex Resource load

### ISSUE THREAD

**Issue by nobuyukinyuu** [godotengine/godot#21590]
**Godot version:**
1ff772b

**OS/device including version:**
Windows 10 x64

**Issue description:**
I'm assuming this might have something to do with how the new inspector works.  Upon attempting to load a large `TileSet` resource for a tilemap, the console gets spammed ad infinitum with the error message "Object was deleted while awaiting a callback" and the editor becomes unresponsive.  This particular tileset resource doesn't always make 3.0.6 very happy either, but although it makes the editor stall for input when loading, it is still usable there.

This tileset can be loaded at runtime without an issue, but attempting to inspect the resource at design time will result in this soft lock.  Any node instances which utilize the tileset as a subresource become impossible to inspect at design time as well.

**Steps to reproduce:**
Create a new project and extract the contents of the archive to root.  Attempt to load the resource.

**Minimal reproduction project:**
[complexTileset.zip](https://github.com/godotengine/godot/files/2334332/complexTileset.zip)

**Comment by coppolaemilio** [comment]
I also had this error (`Object was deleted while awaiting a callback`) when resizing a ColorRect Node, this was the structure:

- Node2D
    - CanvasLayer
        - Control
            - Control
                - **ColorRect**

I was modifying the size x value of that Rect.
After a minute throwing that error the flood stopped and got this

```
TOTAL BYTES: 1048560
NULL count: 0
CALL : 43684
CALL _update_callback: 3
CALL show_modal: 1
CALL grab_focus: 1
CALL select_all: 1
ERROR: Message queue out of memory. Try increasing 'message_queue_size_kb' in project settings.
   At: core/message_queue.cpp:56
```

Godot: v3.1.alpha.official

**Comment by StraToN** [comment]
I am affected by this problem as well, also in simple scenes. I can't determine a common scheme that allows an easy reproduction of this bug. I'll push my repo and link it on this issue.

**Comment by nobuyukinyuu** [comment]
@straton

What kind of assets were they?  Was there anything unusual about them?

**Comment by StraToN** [comment]
@nobuyukinyuu Nothing special. All were from Kenney's assets, both 2D and 3D stuff - but many of them.

**Comment by kubecz3k** [comment]
I personally have an infinite loop of this messages randomly when I'm starting a game/scene from the editor. When it happens it's linked with editor freeze and rapid memory leak.

**Comment by PLyczkowski** [comment]
Removing project folder from Roaming\Godot\projects fix the problem for me.

**Comment by akien-mga** [comment]
Should be fixed by #24194.
